src/NodeParser/StringTemplateLiteralNodeParser.ts

```diff
@@ -18,6 +20,14 @@ export class StringTemplateLiteralNodeParser implements SubNodeParser {
         if (node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral) {
             return new LiteralType(node.text);
         }
+        if (
+            node.templateSpans
+                .map((span) => this.childNodeParser.createType(span.type, context))
+                .some((type) => type instanceof NumberType)
```

Comment:
Somehow this feels restrictive. Why do we special case only numbers?

Reply:
So it seems like below we deal with literals but anything that can't be converted into one should probably fall back to string here. Maybe we can handle the more general case.

Reply:
Yeah, you're right, no reason to only check for numbers here.